### PR TITLE
Handle synchronous events for rolling upgrades.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Add handling for synchronous events, related to #600.   However, I can't get replication_upgrade to get past the following:

```
19:35:42.272 [info] Wait until ring converged on ['dev4@127.0.0.1','dev5@127.0.0.1','dev6@127.0.0.1']
```

@kellymclaughlin do you mind running?

### CODE REVIEW

**Comment by kellymclaughlin** [comment]
I have determined the issue plaguing the `replication_upgrade` test was the [cancellation of fullsync](https://github.com/basho/riak_test/blob/master/tests/replication.erl#L531) when an attempt to start and wait for fullsync completion failed. This step is actually unnecessary because the only reason this condition occurs is that fullsync simply fails to start on the initial call to `riak_repl_console:start_fullsync/1` due to a race condition. So it is never the case that fullsync needs to be canceled. [This PR](https://github.com/basho/riak_test/pull/639) is open to remove that cancellation ad using the changes represented the `replication_upgrade` test passes for me. The PR contains an additional change to replace the wait function used to determine when a replication connection has been established after a node upgrade. It is not necessary for the test to pass, but should decrease the test execution time.

We should investigate why attempting to cancel fullsync when it is not running causes this problem, but in my view that seems like a separate and less severe issue.

File: src/riak_repl_keylist_server.erl [hunk 1]
```diff
@@ -156,64 +161,26 @@ wait_for_partition({start_fullsync, _} = Command, State) ->
...
+    wait_for_individual_partition(Partition, State);
 %% Unknown event (ignored)
 wait_for_partition(Event, State) ->
     lager:debug("Full-sync with site ~p; ignoring event ~p",
         [State#state.sitename, Event]),
     {next_state, wait_for_partition, State}.
 
-build_keylist(Command, State)
-        when Command == cancel_fullsync ->
+build_keylist(Command, State) when Command == cancel_fullsync ->
```
Comment (kellymclaughlin) [inline src/riak_repl_keylist_server.erl:174]: I know this is preexisting cruft, but the guards here could be eliminated in favor an explicit pattern match on the `Command` atoms and would make the code a bit more clear and concise.

Comment (kellymclaughlin) [inline src/riak_repl_keylist_server.erl:174]: Looking closer I see that a similar pattern is used throughout this module so perhaps it is not worth worrying about at this time.

**Comment by kellymclaughlin** [comment]
These changes look good and in concert with the riak_test PR mentioned above it resolves the `replication_upgrade` test failure.

:+1:

**Comment by cmeiklejohn** [comment]
@borshop merge

**Pull request merged**